src/services/window.ts

```diff
@@ -46,6 +48,50 @@ ipcMain.on("create-extra-window", (event, arg) => {
     createExtraWindow(arg.extraWindowHash);
 });
 
+interface ExportAsPdfOpts {
+    title: string;
+    landscape: boolean;
+}
+
+ipcMain.on("export-as-pdf", async (e, opts: ExportAsPdfOpts) => {
+    const browserWindow = BrowserWindow.fromWebContents(e.sender);
+    if (!browserWindow) {
+        return;
+    }
+
+    const filePath = dialog.showSaveDialogSync(browserWindow, {
+        defaultPath: formatDownloadTitle(opts.title, "file", "application/pdf"),
+        filters: [
+            {
+                name: t("pdf.export_filter"),
+                extensions: [ "pdf" ]
+            }
+        ]
+    });
+    if (!filePath) {
+        return;
+    }
+
+    let buffer: Buffer;
+    try {
+        buffer = await browserWindow.webContents.printToPDF({
+            landscape: opts.landscape
+        });
+    } catch (e) {
+        dialog.showErrorBox(t("pdf.unable-to-export-title"), t("pdf.unable-to-export-message"));
+        return;
+    }
+
+    try {
+        fs.writeFileSync(filePath, buffer);
```

Comment:
could we use the async `fs.writeFile` here instead?

Reply:
I wasn't even aware of `fs/promises`, cool thing.